felix/rules/policy.go

```diff
@@ -157,7 +158,14 @@ func filterNets(mixedCIDRs []string, ipVersion uint8, isNegated bool) (filtered
 // isCatchAllCIDR returns true if the CIDR represents "all addresses" for the given IP version.
 // This is used to detect problematic negated matches that would create logical contradictions.
 func isCatchAllCIDR(cidr string, ipVersion uint8) bool {
-	return (ipVersion == 4 && cidr == "0.0.0.0/0") || (ipVersion == 6 && cidr == "::/0")
```

Comment:
Similarly here this isn't based on the IP pool CIDR, so I think we can trust that it is normalized. I am not sure that adding this ParseCIDR into the mainline path here will be worthwhile given potential performance impacts.